src/tests/ftest/container/fill_destroy_loop.py

```diff
@@ -22,53 +24,122 @@ class BoundaryPoolContainerSpace(TestWithServers):
 
     DER_NOSPACE = "-1007"
 
+    def __init__(self, *args, **kwargs):
+        """Initialize a BoundaryPoolContainerSpace object."""
+        super().__init__(*args, **kwargs)
+
+        self.test_loop = 0
+        self.reclaim_props = []
+        self.delta_bytes = 0
+
+    def setUp(self):
+        """Set up each test case."""
+        super().setUp()
+
+        self.test_loop = self.params.get("test_loop", "/run/pool/*", 0)
+        self.reclaim_props = self.params.get("reclaim_props", "/run/pool/*", [])
+
+        delta = self.params.get("delta", "/run/pool/*", "0")
+        self.delta_bytes = human_to_bytes(delta)
+        self.log.info("==> Set pool delta to %s (%i bytes)", delta, self.delta_bytes)
+
+    def check_server_logs(self):
+        """Check if GC engine errors have occurred during the test
+
+        This method parse the engine log to find silent errors regarding the GC engine operations.
+        When the SCM storage is completely full, it is not possible to perform some operations on
+        metadata such as destroying a pool.  Indeed, such operations require some temporary SCM
+        storage to reorganize the B-trees used for recording the data layout of the pools.  When
+        such error occurs, some storage leakage could eventually happen.  At this time, the only
+        reliable way to detect these errors is to check if errors regarding B-tree management have
+        occurred.  With this test, this can be done with looking for error messages generated with
+        the call of the function btr_rec_free().
+        """
+        self.log.info("==>Checking server logs of hosts %s", self.hostlist_servers)
+
+        err_regex = r"'^.+ tree ERR .+ btr_rec_free.+ Failed to free rec:.+$'"
```

Comment:
Checking "btr_rec_free" looks not reliable to me, I'm wondering if we can just simply check DER_NOSPACE error?

Reply:
The issue is that some DER_NOSPACE errors are expected in the test as we are filling the pool.
I am going to check if there is some specific DER_NOSPACE errors in the logs which are only related to GC.

Reply:
- [ ] Grep GC ENOSPACE related error messages

Reply:
Fixed with commit https://github.com/daos-stack/daos/pull/13013/commits/270805402e811b5f153193363cf4304b7dcc26f0#diff-aa5a03545fb0ad9ec37f79bb9c06e107ded1fee8997c345356dcbd6d51c42fca